From 40672c0dfc569099f682f38078497768838c9fe5 Mon Sep 17 00:00:00 2001
From: Osamu Aoki <osamu@debian.org>
Date: Sat, 8 Feb 2014 16:54:20 +0900
Subject: [PATCH] pass multiarch path to -DCMAKE_INSTALL_LIBDIR

== pros ==

This use of CMAKE_INSTALL_LIBDIR seems to be correct solution as suggested by
http://bugs.debian.org/722697  This seems good solution as I read the source.

http://www.cmake.org/cmake/help/v2.8.12/cmake.html#module:GNUInstallDirs
GNUInstallDirs: Define GNU standard installation directories

Provides install directory variables as defined for GNU software:

  http://www.gnu.org/prep/standards/html_node/Directory-Variables.html
Inclusion of this module defines the following variables:

  CMAKE_INSTALL_<dir>      - destination for files of a given type
  CMAKE_INSTALL_FULL_<dir> - corresponding absolute path
where <dir> is one of:
 ...
  LIBDIR  - object code libraries (lib or lib64 or lib/<multiarch-tuple> on Debian)
 ...
Each CMAKE_INSTALL_<dir> value may be passed to the DESTINATION options of
install() commands for the corresponding file type. If the includer does not
define a value the above-shown default will be used and the value will appear
in the cache for editing by the user. Each CMAKE_INSTALL_FULL_<dir> value
contains an absolute path constructed from the corresponding destination by
prepending (if necessary) the value of CMAKE_INSTALL_PREFIX.

== cons ==

It does not work on some packages (ibus-qt) if no change is made.

In order this to work, following patch was required in upstream source:

| --- a/src/CMakeLists.txt
| +++ b/src/CMakeLists.txt
| @@ -104,7 +104,7 @@
|  )
| 
|  install(TARGETS ibus-qt
| -    LIBRARY DESTINATION ${LIBDIR})
| +    LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR})
|  install(FILES ${ibus_qt_HDRS}
|      DESTINATION include/ibus-qt COMPONENT Devel)
| ---
|  Debian/Debhelper/Buildsystem/cmake.pm | 1 +
|  1 file changed, 1 insertion(+)

diff --git a/Debian/Debhelper/Buildsystem/cmake.pm b/Debian/Debhelper/Buildsystem/cmake.pm
index db971db..133e558 100644
--- a/Debian/Debhelper/Buildsystem/cmake.pm
+++ b/Debian/Debhelper/Buildsystem/cmake.pm
@@ -42,6 +42,7 @@ sub configure {
 
 	# Standard set of cmake flags
 	push @flags, "-DCMAKE_INSTALL_PREFIX=/usr";
+	push @flags, "-DCMAKE_INSTALL_LIBDIR=lib/$multiarch";
 	push @flags, "-DCMAKE_VERBOSE_MAKEFILE=ON";
 	push @flags, "-DCMAKE_BUILD_TYPE=None";
 
-- 
1.9.rc1

